source/Patches/Roles/Role.cs

```diff
@@ -512,6 +512,45 @@ public static class HudManager_Update
             private static Vector3 oldScale = Vector3.zero;
             private static Vector3 oldPosition = Vector3.zero;
 
+            //add the colorType List from DeadBody.cs of MedicMod folder to use here
+            public static String GetColorType(PlayerControl player)
+            {
+                var colors = new Dictionary<int, string>
```

Comment:
Initialize this dictionary statically once instead of every time this function is called.

Reply:
I used the original code,
I have to take it out of the function with
private static ....?

Reply:
Yes, take the dictionary outside the function itself. It's really inefficient to keep creating and garbage collecting this function.

Reply:
Like i make in the last commit ?

Reply:
Yes